src/visitors/semantic/symbols/mod.rs

```diff
@@ -0,0 +1,198 @@
+                                &param.loc,
+                            ),
+                        );
+                    }
+                } else {
+                    (*state).symbols.push(SymbolInformation::new(
+                        SymbolKind::Variable,
+                        va.id.name.clone(),
+                        uri.clone(),
+                        node.loc(),
+                    ))
+                }
+            }
+            Node::MemberExpr(me) => {
+                if let Some(source) = me.loc.source.clone() {
+                    (*state).symbols.push(SymbolInformation::new(
+                        SymbolKind::Object,
+                        source.clone(),
+                        uri.clone(),
+                        &me.loc,
+                    ))
+                }
+            }
+            Node::CallExpr(c) => {
+                if let Expression::Identifier(ident) =
+                    c.callee.clone()
+                {
+                    (*state).symbols.push(SymbolInformation::new(
+                        SymbolKind::Function,
+                        ident.name.clone(),
+                        uri.clone(),
+                        &c.loc,
+                    ))
+                }
+
+                for arg in c.arguments.clone() {
+                    if let Expression::Function(_) = arg.value {
+                        (*state).symbols.push(
+                            SymbolInformation::new(
+                                SymbolKind::Function,
+                                arg.key.name.clone(),
+                                uri.clone(),
+                                &arg.loc,
+                            ),
+                        );
+                    } else {
+                        (*state).symbols.push(
+                            SymbolInformation::new(
+                                SymbolKind::Variable,
+                                arg.key.name.clone(),
+                                uri.clone(),
+                                &arg.loc,
+                            ),
+                        );
+                    }
+                }
+            }
+            Node::FloatLit(num) => {
+                (*state).symbols.push(SymbolInformation::new(
+                    SymbolKind::Number,
+                    num.value.to_string(),
+                    uri.clone(),
+                    &num.loc,
+                ));
+                return false;
+            }
+            Node::IntegerLit(num) => {
+                (*state).symbols.push(SymbolInformation::new(
+                    SymbolKind::Number,
+                    num.value.to_string(),
+                    uri.clone(),
+                    &num.loc,
+                ));
+                return false;
+            }
+            Node::DateTimeLit(d) => {
+                (*state).symbols.push(SymbolInformation::new(
+                    SymbolKind::Constant,
```

Comment:
dateTimeLit shouldn't be constant in my option,

Reply:
@kelwang There isn't a better option to represent a literal value